sdk/core/System.Net.ClientModel/api/System.Net.ClientModel.net6.0.cs

```diff
@@ -0,0 +1,407 @@
+        public bool Contains(System.Collections.Generic.KeyValuePair<TKey, TValue> item) { throw null; }
+        public bool ContainsKey(TKey key) { throw null; }
+        public void CopyTo(System.Collections.Generic.KeyValuePair<TKey, TValue>[] array, int arrayIndex) { }
+        public System.Collections.Generic.IEnumerator<System.Collections.Generic.KeyValuePair<TKey, TValue>> GetEnumerator() { throw null; }
+        public bool Remove(System.Collections.Generic.KeyValuePair<TKey, TValue> item) { throw null; }
+        public bool Remove(TKey key) { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+        public bool TryGetValue(TKey key, out TValue value) { throw null; }
+    }
+    public partial class OptionalList<T> : System.Collections.Generic.ICollection<T>, System.Collections.Generic.IEnumerable<T>, System.Collections.Generic.IList<T>, System.Collections.Generic.IReadOnlyCollection<T>, System.Collections.Generic.IReadOnlyList<T>, System.Collections.IEnumerable
+    {
+        public OptionalList() { }
+        public OptionalList(System.Net.ClientModel.Internal.OptionalProperty<System.Collections.Generic.IList<T>> optionalList) { }
+        public OptionalList(System.Net.ClientModel.Internal.OptionalProperty<System.Collections.Generic.IReadOnlyList<T>> optionalList) { }
+        public int Count { get { throw null; } }
+        public bool IsReadOnly { get { throw null; } }
+        public bool IsUndefined { get { throw null; } }
+        public T this[int index] { get { throw null; } set { } }
+        public void Add(T item) { }
+        public void Clear() { }
+        public bool Contains(T item) { throw null; }
+        public void CopyTo(T[] array, int arrayIndex) { }
+        public System.Collections.Generic.IEnumerator<T> GetEnumerator() { throw null; }
+        public int IndexOf(T item) { throw null; }
+        public void Insert(int index, T item) { }
+        public bool Remove(T item) { throw null; }
+        public void RemoveAt(int index) { }
+        public void Reset() { }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public static partial class OptionalProperty
+    {
+        public static bool IsCollectionDefined<T>(System.Collections.Generic.IEnumerable<T> collection) { throw null; }
+        public static bool IsCollectionDefined<TKey, TValue>(System.Collections.Generic.IDictionary<TKey, TValue> collection) { throw null; }
+        public static bool IsCollectionDefined<TKey, TValue>(System.Collections.Generic.IReadOnlyDictionary<TKey, TValue> collection) { throw null; }
+        public static bool IsDefined(object value) { throw null; }
+        public static bool IsDefined(string value) { throw null; }
+        public static bool IsDefined(System.Text.Json.JsonElement value) { throw null; }
+        public static bool IsDefined<T>(T? value) where T : struct { throw null; }
```

Comment:
Is it intentional to use the opposite phrasing from the types above (`IsUndefined`)

Reply:
Anything in the .Internal namespace we are not planning to ship GA, so no need to spend a lot of time on these.  They were pulled out of internal shared source to unblock the demo but will be hidden/reworked before we're done.